src/addr.rs

```diff
@@ -529,6 +580,22 @@ impl Sub<PhysAddr> for PhysAddr {
     }
 }
 
+/// Checks whether the address is canonical.
+
+/// Returns an `Option<u64>` containing the queried address representing if
+/// the address is canonical or not.
+#[inline]
+const fn is_canonical(addr: u64) -> Option<u64> {
```

Comment:
Should I public this like the others? I assumed this should mostly be hidden behind the VirtAddr abstraction.

Reply:
I agree that this method should not be public. The way for users of this crate to check for canonical addresses should be `VirtAddr::try_new(addr: u64) -> Result<VirtAddr>`. However, right now that doesn't work right now (see #299). So we still need this _private_ helper function, but it might make things easier if it had the following signature:
```rust
const fn from_canonical(addr: u64) -> Option<VirtAddr>
```